src/Sylius/Component/Core/Promotion/Applicator/UnitsPromotionAdjustmentsApplicator.php

```diff
@@ -82,4 +90,13 @@ private function addAdjustment(PromotionInterface $promotion, OrderItemUnitInter
 
         $unit->addAdjustment($adjustment);
     }
+
+    private function calculate(int $itemTotal, int $minimumPrice, int $promotionAmount): int
```

Comment:
And now, I'm 100% sure it should be separated service

Reply:
i had different service to handle calculations, but @GSadee said its a BC break so i decided to leave it in private methods

Reply:
Looking at your PR and #13325 it appears to me, that some extraction of this logic would be beneficial. Even in your PR it is present twice and has exactly same reason to exist and change

Reply:
I agree that extraction would be recommended, but in what way to avoid breaking a BC promise?